pkg/controlplane/control/manager.go

```diff
@@ -144,6 +144,99 @@ type Manager struct {
 	logger *logrus.Entry
 }
 
+// Restart coredns pods
+func coreDnsRestart(ctx context.Context, m *Manager) error {
```

Comment:
I would prefer to do this using CoreDNS deployment changes and not by deleting individual Pods, if possible.
The deployment has a specific name so you can just get it directly, instead of iterating over all pods in the namespace.
Looking to do the equivalent of ` kubectl rollout restart deployment...`

Reply:
I tried to perform similar to the suggestion in this [link](https://stackoverflow.com/questions/61335318/how-to-restart-a-deployment-in-kubernetes-using-go-client):

1. retrieve `coredns` deployment
2. update it with `restartedAt` annotation
3. call client.Update

It fails to retrieve the deployment with the below error:

```
ERROR  [2024-06-23 15:47:09] failed to configure CoreDNS: no kind is registered for the type v1.Deployment in scheme
"pkg/runtime/scheme.go:100".  component=controlplane.control.manager file=/home/weit/clusterlink/pkg/
controlplane/control/manager.go line=266
```

I can see that operator main() adds all object types to scheme [here](https://github.com/aviweit/clusterlink/blob/corednsrewrite/cmd/cl-operator/main.go#L60) and [here](https://github.com/kubernetes/client-go/blob/master/kubernetes/scheme/register.go#L84).

Reply:
I managed.